### PR TITLE
CORTX-29079 fault_k8s to main merge

### PR DESCRIPTION
**Pull request by ArchanaLimaye**
# Problem Statement
- PR to merge fault_k8s into main branch

# Design
-  Merge contains changes for:  F-27, F-24E, support bundle , bug fixes  etc

# Coding
-  [ ] Coding conventions are followed and code is consistent

# Testing 
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Review Checklist 
- [ ] PR is self reviewed
- [ ] JIRA number/GitHub Issue added to PR
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained
- [ ] Is there a change in filename/package/module or signature? [Y/N]: 
- [ ] If yes for above point, is a notification sent to all other cortx components? [Y/N]
- [ ] Side effects on other features (deployment/upgrade)? [Y/N]
- [ ] Dependencies on other component(s)? [Y/N]
-     If yes for above point, post link to the corresponding PR.

# Review Checklist 
- [ ] Is perfline test run and the report with and without the changes updated in the PR? [Y/N]: 

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Review by mariyappanp — APPROVED** [review]

**Review by mukhtar-inamdar — APPROVED** [review]

**Review by gauri-bhosale — APPROVED** [review]

**Review by Madhura-08 — APPROVED** [review]

**Comment by mukul-seagate11** [comment]
retest this please

**Review by akash2144 — APPROVED** [review]

**Comment by mukul-seagate11** [comment]
DCO workflow is disabled for now as an exception to unblock & merge the PR

**Comment by ArchanaLimaye** [comment]
As discussed with Neerav, PR will be merged as-is into main. Any alex, codacy issues will be addressed by a separate PR

**Review by shaileshsvc — APPROVED** [review]
Approved

File: ha/util/health_generator/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            "resource_status": "online", # Resource status, recovering, online, failed, unknown, degraded, repairing, repaired, rebalancing, offline, etc.
+            "specific_info": {} # Key-value pairs e.g. "generation_id": "xxxx"
+        },
+         # Repeat the dictionary above with specific values if multiple events to be sent.
+    },
+    "delay": xxxx # If present this will add delay of specified seconds between the events.
+}
+
+# Restrictions
+1. This script needs to be executed inside HA container.
```
[the remark this answers is not in the document]
Comment (ArchanaLimaye) [reply ha/util/health_generator/README.md:55]: Will be addressed in a separate PR

**Pull request merged**